### PR TITLE
BUG: fixes PyPy build (do not use _PyHASH_Imag)

### LINKED ISSUE #42355: BUG: build failure on PyPy in hash functions
The PyPy builds on conda-forge are failing with 

```
    In file included from pandas/_libs/algos.c:663:
    pandas/_libs/src/klib/khash_python.h: In function 'complexobject_hash':
    pandas/_libs/src/klib/khash_python.h:282:38: error: '_PyHASH_IMAG' undeclared (first use in this function)
      282 |     Py_uhash_t combined = realhash + _PyHASH_IMAG * imaghash;
          |                                      ^~~~~~~~~~~~
    pandas/_libs/src/klib/khash_python.h:282:38: note: each undeclared identifier is reported only once for each function it appears in
```

This seems to be related to https://github.com/pandas-dev/pandas/pull/41988 cc @realead

### PR DESCRIPTION
**Pull request by realead**
- [x] closes #42355
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] what's new entry

`_PyHASH_Imag` is not defined in PyPy, thus we define and use our own `_PandasHASH_Imag`.

### CODE REVIEW

**Comment by jreback** [comment]
@realead can you add a release note in 1.3.1; ping on green.

File: doc/source/whatsnew/v1.3.1.rst [hunk 1]
```diff
@@ -14,7 +14,7 @@ including other versions of pandas.
 
 Fixed regressions
 ~~~~~~~~~~~~~~~~~
--
+- Pandas could not be build on PyPy (GH-42355)
```
Comment (jreback) [inline doc/source/whatsnew/v1.3.1.rst:17]:
```suggestion
- Pandas could not be built on PyPy (:issue:`42355`)
```

**Comment by jreback** [comment]
thanks @realead

**Comment by jreback** [comment]
@meeseeksdev backport 1.3.x

**Comment by jorisvandenbossche** [comment]
@realead Thanks for the quick fix!

**Pull request merged**